tokman/app.py

```diff
@@ -0,0 +1,115 @@
+# Copyright Contributors to the Packit project.
+# SPDX-License-Identifier: MIT
+
+import logging
+import os
+
+from datetime import datetime
+from pathlib import Path
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+from github import GithubIntegration
+
+try:
+    from flask_restx import Api, Resource
+except ModuleNotFoundError:
+    from flask_restplus import Api, Resource
+
+
+api = Api()
+db = SQLAlchemy()
+github_integration = None
+token_renew_at = None
+
+
+def create_app():
+    app = Flask(__name__)
+    app.config.from_envvar("TOKMAN_CONFIG")
+
+    log_level = os.getenv("LOG_LEVEL", "info")
+    log_level = getattr(logging, log_level.upper())
+    logging.basicConfig(level=log_level)
+
+    private_key = Path(app.config["GITHUB_APP_PRIVATE_KEY"]).read_text()
+    app_id = int(app.config["GITHUB_APP_ID"])
+    global github_integration
+    github_integration = GithubIntegration(app_id, private_key)
+    global token_renew_at
+    token_renew_at = int(app.config.get("TOKEN_RENEW_AT", 60))
+
+    api.init_app(app)
+    db.init_app(app)
+
+    return app
+
+
+class Token(db.Model):
+    __tablename__ = "tokens"
+    id = db.Column(db.Integer, primary_key=True)
+    namespace = db.Column(db.String(), unique=True, nullable=False)
+    repository = db.Column(db.String(), unique=True, nullable=False)
+    token = db.Column(db.String(), unique=True, nullable=True)
+    expires_at = db.Column(db.DateTime, nullable=True)
+
+    def is_expired(self):
+        return (
+            self.expires_at is None
+            or self.token is None
+            or (self.expires_at - datetime.utcnow()).seconds < token_renew_at
+        )
```

Comment:
Can we split this file a bit to have some structure from the start?

- models/database handling
- implementation
- api

Reply:
I'm not against it, though I don't see as a must right now, as I don't expect this app to grow in size anytime soon.

Reply:
Looks like KISS so far, imo could be split later _when necessary_.